astropy/modeling/tests/test_fitters.py

```diff
@@ -654,6 +654,82 @@ def test_with_fitters_and_sigma_clip(self):
+        self.z1d = self.z.flatten()
+        self.weights1d = self.weights.flatten()
+
+    def test_1d_without_weights_without_sigma_clip(self):
+        model = models.Polynomial1D(0)
+        fitter = LinearLSQFitter()
+        fit = fitter(model, self.x1d, self.z1d)
+        assert_allclose(fit.parameters[0], self.z1d.mean(), atol=10**(-2))
+
+    def test_1d_without_weights_with_sigma_clip(self):
+        model = models.Polynomial1D(0)
+        fitter = FittingWithOutlierRemoval(LinearLSQFitter(), sigma_clip,
+                                           niter=3, sigma=3.)
+        filtered, fit = fitter(model, self.x1d, self.z1d)
+        assert(filtered.count() == self.z1d.size - 2)
+        assert(filtered.mask[0] and filtered.mask[1])
+        assert_allclose(fit.parameters[0], 0.0, atol=10**(-2)) # with removed outliers mean is 0.0
+
+    def test_1d_with_weights_without_sigma_clip(self):
+        model = models.Polynomial1D(0)
+        fitter = LinearLSQFitter()
+        fit = fitter(model, self.x1d, self.z1d, weights=self.weights1d)
+        assert(fit.parameters[0] > 1.0)     # outliers pulled it high
+
+    def test_1d_with_weights_with_sigma_clip(self):
+        """smoke test for #7020 - fails without fitting.py patch because weights does not propagate"""
+        model = models.Polynomial1D(0)
+        fitter = FittingWithOutlierRemoval(LinearLSQFitter(), sigma_clip,
+                                           niter=3, sigma=3.)
+        filtered, fit = fitter(model, self.x1d, self.z1d, weights=self.weights1d)
+        assert(fit.parameters[0] > 10**(-2))  # weights pulled it > 0
+        assert(fit.parameters[0] < 1.0)       # outliers didn't pull it out of [-1:1] because they had been removed
+
+    def test_2d_without_weights_without_sigma_clip(self):
+        model = models.Polynomial2D(0)
+        fitter = LinearLSQFitter()
+        fit = fitter(model, self.x, self.y, self.z)
+        assert_allclose(fit.parameters[0], self.z.mean(), atol=10**(-2))
+
+    def test_2d_without_weights_with_sigma_clip(self):
+        model = models.Polynomial2D(0)
+        fitter = FittingWithOutlierRemoval(LinearLSQFitter(), sigma_clip,
+                                           niter=3, sigma=3.)
+        filtered, fit = fitter(model, self.x, self.y, self.z)
+        assert(filtered.count() == self.z.size - 2)
+        assert(filtered.mask[0,0] and filtered.mask[0,1])
+        assert_allclose(fit.parameters[0], 0.0, atol=10**(-2))
+
+    def test_2d_with_weights_without_sigma_clip(self):
+        model = models.Polynomial2D(0)
+        fitter = LevMarLSQFitter() # LinearLSQFitter doesn't handle weights properly in 2D
+        fit = fitter(model, self.x, self.y, self.z, weights=self.weights)
+        assert(fit.parameters[0] > 1.0)     # outliers pulled it high
+
+    def test_2d_with_weights_with_sigma_clip(self):
+        """smoke test for #7020 - fails without fitting.py patch because weights does not propagate"""
```

Comment:
You could use something more direct here, like:
"This test verifies that weights correctly propagate with `FittingWithOutlierRemoval`".